Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -122,6 +122,22 @@ The score is defined as follows and can be used for a password strength bar for
 4 # very unguessable: strong protection from offline slow-hash scenario. (guesses >= 10^10)
 ```
 
+Note that zxcvbn can be implemented by the app-developer as well using the Java (or other) implementation in order to guide the user into suing a strongpassword.
+
+#### Have I Been Pwned: PwnedPasswords
+
+In order to further reduce the likelihood of a succesfull dictionary attack against a sinlge factor authentication scheme (e.g. password only), you can verify whether a password has been used too often. This can be done using services offered through api.pwnedpasswords.com based on [Have i been pwned?](https://haveibeenpwned.com "';--have i been pwned?") as setup by Troy Hunt.
```

Comment:
+ little rephrase, what do you think?

```suggestion
In order to further reduce the likelihood of a successful dictionary attack against a single factor authentication scheme (e.g. password only), you can verify whether a password has been compromised in a data breach. This can be done using services based on the Pwned Passwords API by Troy Hunt (available at api.pwnedpasswords.com). For example, the "[Have I been pwned?](https://haveibeenpwned.com "';--have i been pwned?")" companion website.
```